services/monitor/Dockerfile

```diff
@@ -0,0 +1,22 @@
+FROM alpine:latest
+
+RUN apk add --no-cache \
+  bash \
+  curl \
+  docker-cli \
+  docker-cli-compose \
+  ca-certificates
+
+# More explicit user creation with better defaults
+RUN addgroup -g 1000 monitor && \
+  adduser -D -u 1000 -G monitor -s /bin/bash monitor
+
+COPY --chown=monitor:monitor --chmod=755 monitor.sh /usr/local/bin/monitor.sh
+WORKDIR /workspace
+
+# Basic health check for reliability
```

Comment:
Great attention to detail, watching the watchers :D
Is the container restarted by Docker itself, docker-compose or Kubernetes? Or something else?